crypto/src/util/eq.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::BitXor;
+
+pub enum SliceEqualityCheckMethod {
+    Normal,
+    TimingResistant,
+}
+
+impl SliceEqualityCheckMethod {
+    pub fn timing_resistant_equal<T: Eq + BitXor + Copy>(a: &[T], b: &[T]) -> bool
```

Comment:
Is there a specific reason we did this "in house" rather than use something like subtle? https://docs.rs/subtle/2.4.1/subtle/trait.ConstantTimeEq.html

This appears to be fine to me for what it's worth but given we have an off the shelf trustworthy alternative I'm tempted to suggest it.

Reply:
Because every other implementation has its quirks and this is simple enough I don't want to worry about it. For example, the library you linked does a short-circuit on equality if the lengths are not equal, which basically can be considered an attack vector depending on the context.